test/e2e/suite_test.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1alpha2"
+
+	"sigs.k8s.io/kueue/pkg/util/pointer"
+	//+kubebuilder:scaffold:imports
+)
+
+var (
+	k8sClient client.Client
+	ctx       context.Context
+)
+
+func CreateClientUsingCluster() client.Client {
+
+	envTest := &envtest.Environment{
```

Comment:
I addressed your first point.  I already wait until my kueue-controller-manager is ready.  I don't think services have any status to be ready.  I could do a poll on the service but I am unsure how to reliable grab the host and port of the webhook service.  

Is it sufficient to wait until the deployment of kueue is ready?

Reply:
Maybe it's easier to just retry the object creation with backoff?

Reply:
I'll look into a way to do this without sleeps.  Sleeps almost always bit me in the end so I'll see if I can reliably get host/port so I can poll the webserver.

Other option is to use Eventually on the Create objects.  Once they successfully complete, I think that would mean that the webhooks are working?

Reply:
I added the eventually step to testing if Kueue is ready. I think this should be enough as it will keep calling webhook of resource flavor until it passes.